src/Features/Traits/Dispatcher.php

```diff
@@ -4,6 +4,97 @@
+use Illuminate\Support\Testing\Fakes\BusFake;
+use Illuminate\Support\Testing\Fakes\EventFake;
+use InvalidArgumentException;
+
 trait Dispatcher
 {
+    use Application;
+
+    private BusFake $busFake;
+    private EventFake $eventFake;
+
+    /**
+     * @Given bus is running
+     * @throws BindingResolutionException
+     */
+    public function runBus(): void
+    {
+        $this->busFake = new BusFake($this->getContainer()->make(BusDispatcher::class));
+    }
+
+    /**
+     * @Given events are faked
+     * @throws BindingResolutionException
+     */
+    public function fakeEvents(): void
+    {
+        $this->eventFake = new EventFake($this->getContainer()->make(EventDispatcher::class));
+    }
+
+    /**
+     * @When I dispatch :count :objectName jobs with parameters:
+     * @When I dispatch :count :objectName jobs
+     * @when I dispatch :objectName job
+     * @When I dispatch :count of :objectName events with parameters:
+     * @When I dispatch :count of :objectName events
+     * @when I dispatch :objectName event
+     * @When I dispatch :count of :objectName with parameters:
+     * @When I dispatch :count of :objectName
+     * @when I dispatch :objectName
+     */
+    public function dispatchObject(string $objectName, ?TableNode $table = null, int $count = 1): void
+    {
+        $parameters = [];
+
+        if ($table) {
+            foreach ($table as $row) {
+                $value = is_numeric($row["value"]) ? (int)$row["value"] : (string)$row["value"];
+                $parameters[] = $value;
+            }
+        }
+
+        $objectClass = RecognizeClassHelper::recognizeObjectClass($objectName);
+
+        $object = new $objectClass(...$parameters);
+
+        for ($i = 0; $i < $count; $i++) {
+            $this->resolveFaker($objectName)->dispatch($object);
+        }
+    }
+
+    /**
+     * @Then I should see :objectName event was dispatched
+     * @Then I should see :count of :objectName events were dispatched
+     * @Then I should see :objectName job was dispatched
+     * @Then I should see :count of :objectName jobs were dispatched
+     * @Then I should see :objectName was dispatched
+     * @Then I should see :count of :objectName were dispatched
+     */
+    public function assertDispatched(string $objectName, int $count = 1): void
+    {
+        $objectClass = RecognizeClassHelper::recognizeObjectClass($objectName);
+
+        $this->resolveFaker($objectName)->assertDispatched($objectClass, $count);
+    }
+
+    private function resolveFaker($objectName): BusFake|EventFake
+    {
+        $objectType = RecognizeClassHelper::guessType($objectName);
+
+        if ($objectType === TypesEnum::JOB->value) {
+            return $this->busFake;
+        } elseif ($objectType === TypesEnum::EVENT->value) {
+            return $this->eventFake;
+        }
+
+        throw new InvalidArgumentException("Unsupported object type: $objectType");
```

Comment:
This could be more verbose as a match:
```suggestion
        return match ($objectType) {
            TypesEnum::JOB->value => $this->busFake,
            TypesEnum::EVENT->value => $this->eventFake,
            default => throw new InvalidArgumentException("Unsupported object type: $objectType"),
        };
```